src/GovUk.Education.ExploreEducationStatistics.Admin/Requests/Public.Data/DataSetVersionCreateRequest.cs

```diff
@@ -0,0 +1,27 @@
+#nullable enable
+using FluentValidation;
+using GovUk.Education.ExploreEducationStatistics.Admin.Services.Interfaces.Public.Data;
+using GovUk.Education.ExploreEducationStatistics.Admin.Validators;
+using GovUk.Education.ExploreEducationStatistics.Common.Extensions;
+using System;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Requests.Public.Data;
+
+public record DataSetVersionCreateRequest
+{
+    public required Guid ReleaseFileId { get; init; }
+
+    public class Validator : AbstractValidator<DataSetVersionCreateRequest>
+    {
+        public Validator(IDataSetVersionRepository dataSetVersionRepository)
+        {
+            RuleFor(request => request.ReleaseFileId).MustAsync(async (releaseFileId, cancellationToken) => 
+                    await dataSetVersionRepository.GetByReleaseFileId(
+                        releaseFileId: releaseFileId,
+                        cancellationToken: cancellationToken)
+                    is null)
```

Comment:
We might not need a repository method to do this as the code in that method is simple and it's also fetching back the data set version, whereas this could make use of `AnyAsync` to check for existence if using the database context directly and that would be more efficient.

Reply:
Moved to the `DataSetVersionService` as a method called `HasExistingVersion`